lib/doctrine.js

```diff
@@ -324,6 +324,10 @@
                     name += scanIdentifier(last);
 
                 }
+                if(source.charCodeAt(index) === 0x5B  /* '[' */ && source.charCodeAt(index + 1) === 0x5D  /* ']' */){
+                    name += advance();
+                    name += advance();
+                }
                 while (source.charCodeAt(index) === 0x2E  /* '.' */ ||
```

Comment:
I tried moving the code inside of the loop, but it broke a couple of tests. I think this is a good start.